OutOfSchool/OutOfSchool.WebApi/Services/ChildService.cs

```diff
@@ -15,124 +18,101 @@ namespace OutOfSchool.WebApi.Services
     /// </summary>
     public class ChildService : IChildService
     {
-        private IEntityRepository<Child> repository;
+        private readonly IEntityRepository<Child> repository;
+        private readonly ILogger logger;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="ChildService"/> class.
         /// </summary>
         /// <param name="repository">Repository for the Child entity.</param>
-        public ChildService(IEntityRepository<Child> repository)
+        /// <param name="logger">Logger.</param>
+        public ChildService(IEntityRepository<Child> repository, ILogger logger)
         {
             this.repository = repository;
+            this.logger = logger;
         }
 
         /// <inheritdoc/>
         public async Task<ChildDTO> Create(ChildDTO dto)
         {
-            if (dto == null)
-            {
-                throw new ArgumentNullException(nameof(dto), "Child was null.");
-            }
-
-            if (dto.DateOfBirth > DateTime.Now)
-            {
-                throw new ArgumentException("Invalid Date of birth");
-            }
+            logger.Information("Child creating was started.");
 
-            try
-            {
-                var child = dto.ToDomain();
+            var child = dto.ToDomain();
 
-                var newChild = await repository.Create(child).ConfigureAwait(false);
+            var newChild = await repository.Create(child).ConfigureAwait(false);
 
-                return newChild.ToModel();
-            }
-            catch (Exception ex)
-            {
-                throw new Exception($"{nameof(dto)} could not be saved: {ex.Message}");
-            }
+            return newChild.ToModel();
         }
 
         /// <inheritdoc/>
         public async Task<IEnumerable<ChildDTO>> GetAll()
```

Comment:
probably we also need Get methods with some filters like ParentId or SectionId

Reply:
Do you mean to get a list of children of a specific workshop, by workshop's id? And the same for parent - get all his/her children?

Reply:
yes